cmd/notary/integration_test.go

```diff
@@ -162,23 +161,50 @@ func splitLines(chunk string) []string {
 	return results
 }
 
-// List keys, parses the output, and returns the keys as an array of root key
-// IDs and an array of signing key IDs
-func GetKeys(t *testing.T, tempDir string) ([]string, []string) {
+// List keys, parses the output, and returns the unique key IDs as an array
+// of root key IDs and an array of signing key IDs.  Output expected looks like:
+//     ROLE      GUN          KEY ID                   LOCATION
+// ----------------------------------------------------------------
+//   root               8bd63a896398b558ac...   file (.../private)
+//   snapshot   repo    e9e9425cd9a85fc7a5...   file (.../private)
+//   targets    repo    f5b84e2d92708c5acb...   file (.../private)
+func getUniqueKeys(t *testing.T, tempDir string) ([]string, []string) {
 	output, err := runCommand(t, tempDir, "key", "list")
 	assert.NoError(t, err)
+	lines := splitLines(output)
 
-	parts := strings.Split(output, "# Signing keys:")
-	assert.Len(t, parts, 2)
-
-	fixed := make([][]string, 2)
-	for i, part := range parts {
-		fixed[i] = splitLines(
-			strings.TrimPrefix(strings.TrimSpace(part), "# Root keys:"))
-		sort.Strings(fixed[i])
+	var (
+		rootMap    = make(map[string]bool)
+		nonrootMap = make(map[string]bool)
+		root       []string
+		nonroot    []string
+	)
+	// first two lines are header
+	for _, line := range lines[2:] {
```

Comment:
Small nit, but what if output is empty, and lines is empty too? Wouldn't lines[2:] fail?